numpy/random/_generator.pyx

```diff
@@ -3535,7 +3535,7 @@ cdef class Generator:
             # if check_valid == 'ignore' we need to ensure that np.sqrt does not
             # return a NaN if s is a very small negative number that is
             # approximately zero or when the covariance is not positive-semidefinite
-            _factor = u * np.sqrt(abs(s))
```

Comment:
Looking at this more carefully, `cov = (u * sqrt(s)) @ (u * sqrt(s)).T` so I think this is correct.

Reply:
Yes this is correct for the covariance (`_factor @ _factor.T = cov`), but then the `np.dot(x, _factor)` below is incorrect. In particular, for the `svd` version, you will realize that `_factor @ factor.T != cov`, but instead  `_factor.T @ factor = cov`.

Assuming that `_factor @ factor.T = cov`, what you want is 
```
E[(_factor @ x.T) @ (_factor @ x.T).T]
= E[_factor @ (x.T @ x) @ _factor.T]
= _factor @ E[x.T @ x] @ factor.T
= _factor @ I @ _factor.T
= cov
```
where for the single sample `x` here is `1 x n` with `n` the size of the covariance matrix. 
So the correlated samples here should be `_factor @ x.T` and not `np.dot(x, _factor) = x @ _factor`, otherwise what you getting instead now is sth like
```
E[(x @ _factor) @ (x @ _factor).T]
= E[x @ (_factor @ _factor.T) @ x.T]
= E[x @ cov @ x.T]
```

So two possible changes: 
1. Change the `_factor` definition for `eigh` as done in this one line diff and keep using `x @ _factor` as the correlated samples
2. Change the `_factor` definition for the `svd` and `cholesky` methods to be consistent s.t. `_factor @ factor.T = cov`, and use `_factor @ x` for the correlated samples throughout.

Reply:
Yeah, I was just thinking I should check the rest of the implementation. The first way is the least intrusive, OTOH, the second is how I think about the problem and may be more easily extensible.  For now, your PR is the simplest fix.